solana/src/lib.rs

```diff
@@ -227,6 +248,73 @@ impl SolanaNetwork for SolanaRpc {
 
         Ok(())
     }
+
+    async fn latest_transaction(&self) -> Result<Signature, Self::Error> {
+        let signer = Keypair::from_bytes(&self.keypair).unwrap();
```

Comment:
this unwrap safe?

Reply:
Yup, we do this in the burn function as well